packages/blockchain-extension/extension/commands/testSmartContractCommand.ts

```diff
@@ -123,43 +123,43 @@ export async function testSmartContract(allContracts: boolean, chaincode?: Insta
     let localSmartContractDirectory: string;
     let testFileSuffix: string;
     let workspaceProjectName: string;
-    if (contractLang === 'golang' ) {
-        outputAdapter.log(LogType.ERROR, `Automated functional tests for a smart contract project written in ${contractLang} are currently not supported.`);
-        return;
+
+    const languagesToSuffixes: object = {
+        JavaScript: 'js',
+        TypeScript: 'ts',
+    };
+    if (contractLang === 'java') {
+        languagesToSuffixes['Java'] = 'java';
+        testLang = 'Java';
+        functionalTestsDirectory = path.join(chosenPackageFolder.uri.fsPath, 'src', 'test', 'java', 'org', 'example');
+    } else if (contractLang === 'golang') {
+        languagesToSuffixes['Golang'] = 'go';
+        testLang = 'Golang';
+        functionalTestsDirectory = chosenPackageFolder.uri.fsPath;
     } else {
-        const languagesToSuffixes: object = {
-            JavaScript: 'js',
-            TypeScript: 'ts',
-        };
-        if (contractLang === 'java') {
-            languagesToSuffixes['Java'] = 'java';
-            testLang = 'Java';
-            functionalTestsDirectory = path.join(chosenPackageFolder.uri.fsPath, 'src', 'test', 'java', 'org', 'example');
-        } else {
-            // Node smart contract: package.json must exist, chaincode and project names must match.
-            const packageJsonFile: string = path.join(chosenPackageFolder.uri.fsPath, 'package.json');
-            const packageJSONBuffer: Buffer = await fs.readFile(packageJsonFile);
-            const packageJSONObj: any = JSON.parse(packageJSONBuffer.toString('utf8'));
-            const replaceRegex: RegExp = /@.*?\//;
-            workspaceProjectName = packageJSONObj.name;
-            workspaceProjectName = workspaceProjectName.replace(replaceRegex, '');
-            if (workspaceProjectName === chaincodeName) {
-                // Smart contract is open in workspace
-                functionalTestsDirectory = path.join(chosenPackageFolder.uri.fsPath, 'functionalTests');
-                localSmartContractDirectory = chosenPackageFolder.uri.fsPath;
-                const testLanguageItem: LanguageQuickPickItem = await UserInputUtil.showLanguagesQuickPick('Choose preferred test language', [], Object.keys(languagesToSuffixes));
-                if (!testLanguageItem) {
-                    return;
-                }
-                testLang = testLanguageItem.label;
-            } else {
-                // Unable to identify the correct project in workspace.
-                outputAdapter.log(LogType.ERROR, `Smart contract project ${chaincodeName} does not correspond to the selected open project ${workspaceProjectName}.`);
+        // Node smart contract: package.json must exist, chaincode and project names must match.
+        const packageJsonFile: string = path.join(chosenPackageFolder.uri.fsPath, 'package.json');
+        const packageJSONBuffer: Buffer = await fs.readFile(packageJsonFile);
```

Comment:
You can just change this to be `await fs.readJson(packageJsonFile)` - that will handle reading the file and parsing it.

Reply:
Just noticed this is code that already exists - don't worry about this then!

Reply:
indeed I tried to keep the current code as is.